pages/start_page.py

```diff
@@ -25,45 +25,48 @@ def sign_up(self, username='', email='', password=''):
 
     def find_elem_create_post(self):
         assert self.driver.find_element(by=By.XPATH,
-                                        value=".//a[@href='/create-post']").is_displayed(), "Not found button 'create-post'"
-
-    # TODO: def find_elem_profile_name(self):
-    #     assert self.driver.find_element(by=By.XPATH, value=f".//a[@href='/profile/{test_username}']").is_displayed()
+                                        value=self.constants.CREATE_POST).is_displayed(), "Not found button 'create-post'"
 
+    @wait_until_ok(timeout=7, period=1)
     def click_sign_up_button(self):
-        sleep(1)
+        self.wait_until_clickable(xpath=self.constants.SIGN_UP_BUTTON)
+        self.find_and_click(xpath=self.constants.SIGN_UP_BUTTON)
+
+    @wait_until_ok(timeout=10, period=1)
+    def click_sign_up_and_verify(self):
         self.click(xpath=self.constants.SIGN_UP_BUTTON)
+        assert not self.is_element_exists(xpath=self.constants.SIGN_UP_BUTTON)
 
     def username_error(self):
-        assert self.driver.find_element(by=By.XPATH,
-                                        value=".//div[contains(text(),'Username must be at least 3 characters.')]").is_displayed(), \
+        assert self.driver.find_element(by=By.XPATH, value=self.constants.SIGN_UP_USERNAME_ERROR).is_displayed(), \
             "Not see message 'Username must be at least 3 characters.'"
 
     def username_error_second(self):
-        assert self.driver.find_element(by=By.XPATH,
-                                        value=".//div[contains(text(),'Username can only contain letters and numbers.')]").is_displayed(), \
+        assert self.driver.find_element(by=By.XPATH, value=self.constants.SIGN_UP_USERNAME_ERROR2).is_displayed(), \
             "Not see message 'Username can only contain letters and numbers.'"
 
     def password_error(self):
-        assert self.driver.find_element(by=By.XPATH,
-                                        value=".//div[contains(text(),'Password must be at least 12 characters.')]").is_displayed(), \
+        assert self.driver.find_element(by=By.XPATH, value=self.constants.SIGN_UP_PASSWORD_ERROR).is_displayed(), \
             "Password must be at least 12 characters."
 
     def email_error(self):
-        assert self.driver.find_element(by=By.XPATH,
-                                        value=".//div[contains(text(),'You must provide a valid email address.')]").is_displayed(), \
+        assert self.driver.find_element(by=By.XPATH, value=self.constants.SIGN_UP_EMAIL_ERROR).is_displayed(), \
             "We not see message 'You must provide a valid email address'"
 
     def display_sign_up_button(self):
-        assert self.driver.find_element(by=By.XPATH,
-                                        value='.//button[contains(text(),"Sign up for OurApp")]').is_displayed(), \
+        assert self.driver.find_element(by=By.XPATH, value=self.constants.SIGN_UP_BUTTON).is_displayed(), \
```

Comment:
Hint: You could add some `is_displayed` method, so would not have to pass `by` each time and verification would looks better
```
assert self.is_displayed(self.constants.SIGN_UP_BUTTON)
```

Reply:
Thx for the hint. Update.